SocketRocket/SRWebSocket.m

```diff
@@ -145,6 +145,122 @@ @implementation SRWebSocket {
     
     NSArray<NSString *> *_requestedProtocols;
     SRIOConsumerPool *_consumerPool;
+
+    //Proxy support
+    NSString *_httpProxyHost;
+    uint32_t _httpProxyPort;
+    BOOL _connectingToProxy;
+
+}
+
+// get proxy setting from device setting
+-(void) configureProxy
+{
+    SRFastLog(@"configureProxy");
+    NSDictionary *proxySettings = CFBridgingRelease(CFNetworkCopySystemProxySettings());
+    NSArray *proxies = CFBridgingRelease(CFNetworkCopyProxiesForURL((__bridge CFURLRef)_url, (__bridge CFDictionaryRef)proxySettings));
+    if (proxies.count == 0) {
+        SRFastLog(@"configureProxy no proxies");
+        [self openConnection];
+        return;                 // no proxy
+    }
+    NSDictionary *settings = [proxies objectAtIndex:0];
+    NSString *proxyType = settings[(NSString *)kCFProxyTypeKey];
+    if ([proxyType isEqualToString:(NSString *)kCFProxyTypeAutoConfigurationURL]) {
+        NSURL *pacURL = settings[(NSString *)kCFProxyAutoConfigurationURLKey];
+        if (pacURL) {
+            [self fetchPAC:pacURL];
+            return;
+        }
+    }
+    if ([proxyType isEqualToString:(NSString *)kCFProxyTypeHTTP] || [proxyType isEqualToString:(NSString *)kCFProxyTypeHTTPS]) {
+        _httpProxyHost = settings[(NSString *)kCFProxyHostNameKey];
+        _httpProxyPort = [settings[(NSString *)kCFProxyPortNumberKey] intValue];
+    }
+    if (_httpProxyHost) {
+        SRFastLog(@"SRWebSocket using proxy %@:%u", _httpProxyHost, _httpProxyPort);
+    }
+    [self openConnection];
+}
+
+- (void)fetchPAC:(NSURL *)PACurl
+{
+    SRFastLog(@"SRWebSocket fetchPAC:%@", PACurl);
+
+	if ([PACurl isFileURL]) {
+        NSError *nsError = nil;
+        NSString *script = [NSString stringWithContentsOfURL:PACurl
+                                                usedEncoding:NULL
+                                                       error:&nsError];
+
+        if (nsError) {
+            [self openConnection];
+            return;
+        }
+        [self runPACScript:script];
+		return;
+	}
+
+	NSString *scheme = [PACurl.scheme lowercaseString];
+    if (![scheme isEqualToString:@"http"] && ![scheme isEqualToString:@"https"]) {
+        // Don't know how to read data from this URL, we'll have to give up
+        // We'll simply assume no proxies, and start the request as normal
+        [self openConnection];
+		return;
+	}
+    __weak typeof(self) weakSelf = self;
+    NSURLRequest *request = [NSURLRequest requestWithURL:PACurl
+                                             cachePolicy:NSURLRequestReturnCacheDataElseLoad
```

Comment:
Is there a specific reason on why you would want to use `CacheDataElseLoad`? If yes - could you please add a comment, so the next person looking at it can figure this out.